pkg/models/dashboards_public.go

```diff
@@ -0,0 +1,43 @@
+package models
+
+var (
+	ErrPublicDashboardFailedGenerateUniqueUid = DashboardErr{
+		Reason:     "Failed to generate unique dashboard id",
+		StatusCode: 500,
+	}
+	ErrPublicDashboardNotFound = DashboardErr{
+		Reason:     "Public dashboard not found",
+		StatusCode: 404,
+		Status:     "not-found",
+	}
+	ErrPublicDashboardIdentifierNotSet = DashboardErr{
+		Reason:     "No Uid for public dashboard specified",
+		StatusCode: 400,
+	}
+)
+
+type PublicDashboardConfig struct {
+	IsPublic        bool            `json:"isPublic"`
+	PublicDashboard PublicDashboard `json:"publicDashboard"`
+}
+
+type PublicDashboard struct {
+	Uid           string `json:"uid" xorm:"uid"`
+	DashboardUid  string `json:"dashboardUid" xorm:"dashboard_uid"`
+	OrgId         int64  `json:"orgId" xorm:"org_id"`
+	TimeVariables string `json:"timeVariables" xorm:"time_variables"`
+}
+
+func (pd PublicDashboard) TableName() string {
+	return "dashboard_public_config"
+}
+
+//
+// COMMANDS
+//
+
+type SavePublicDashboardConfigCommand struct {
+	DashboardUid          string
+	OrgId                 int64
+	PublicDashboardConfig PublicDashboardConfig
```

Comment:
For DTO's, we're following convention of having a Command to do a thing. We need a dashboardUid + orgId in order to query for the dashboard, so those are supplied here. This could be combined with the PublicDashboardConfig as an alternative.

PublicDashboardConfig contains (for now) a single PublicDashboard. In the very near future, this will be an array of PublicDashboard that allows us to expire urls and set specific variables and time ranges on each

Reply:
Ah, this was something very new to me. I thought this would be on the access key level not the public dashboard config level, but I can see this working as well. A design doc or product DNA doc describing this would have been helpful :) 
but I would still have the enable flag on each config or main table with shared config for all access keys, so you can enable/revoke a public dashboard without modifying the main dashboard entity, updating the main dashboard will update its version (optimistic concurrency) which if not also updated in the frontend will prevent further writes, could also cause issues with git update when we introduce git integration for dashboards.